internal/test/wmcb/wmcb_test.go

```diff
@@ -87,11 +82,52 @@ type pkgInfo struct {
 	shaType string
 }
 
+// interface to populate pkInfo structs
+type initPkgInfo interface {
+	// set fields for packageInfo struct
+	initializePkgInfo() error
+	// get checksum file content
+	getSHAFileContent(url string) (string, error)
+	// get SHAvalue for the package
+	getSHAValue(url string) (string, error)
+}
+
 type wmcbFramework struct {
 	// TestFramework holds the instantiation of test suite being executed
 	*e2ef.TestFramework
-	// cniPlugins contains information about the CNI plugin package
-	cniPlugins pkgInfo
+	// contains array of packages to be downloaded
+	pkgs map[string]pkgInfo
+}
+
+// cniPlugins contains information about the CNI plugin package
+type CniPlugins struct {
+	pkgInfo
+}
+
+//kubeNode contains the information about  the kubernetes node package for Windows
+type KubeNode struct{
+	pkgInfo
+}
+
+// initializePackages sets up all the required packages of type PkgInfo
+func (f *wmcbFramework) initializePackages() error{
+	// map to store all required pkgInfo
+	packages:= make(map[string]pkgInfo)
+	cniPluginsPackage := CniPlugins{}
+	kubeNodePackage := KubeNode{}
+	// intialize cni plugin package and add to the package map
+	if err :=  cniPluginsPackage.initializePkgInfo(); err != nil {
+		return fmt.Errorf("unable to initialize CNI Plugins package info: %v", err)
+	}
+	packages["cniPlugins"] = cniPluginsPackage.pkgInfo
+	// initialize kube Node binary package and add to the package map
+	if err := kubeNodePackage.initializePkgInfo(); err != nil{
+		return fmt.Errorf("unable to initialize kubernetes node package info for Windows: %v", err)
+	}
+	packages["kubeNode"] = kubeNodePackage.pkgInfo
```

Comment:
After you create [packages](https://github.com/openshift/windows-machine-config-bootstrapper/pull/170/files#diff-631087f953268c655ad0d05d71b2f6e2R115) you can insert the key and make it tigther for validation.